PhotoStream/app/src/main/java/com/keenanzucker/photostream/DbHelper.java

```diff
@@ -0,0 +1,70 @@
+package com.keenanzucker.photostream;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import java.util.ArrayList;
+
+/**
+ * Created by keenan on 10/2/15.
+ * Database helper used to save images, retrieve the saved images, and delete images from the SQL database
+ * Database is simple table that holds saved image URLs.
+ */
+public class DbHelper{
+
+    private DatabaseService dbs;
+
+    public DbHelper(Context context){
+
+        dbs = new DatabaseService(context);
+    }
+
+    public void saveImage(String url)
+    {
+        // Gets the data repository in write mode
+        SQLiteDatabase db = dbs.getWritableDatabase();
```

Comment:
You'll notice a performance increase if you move this line tot eh constructor and then reference db in this method. That way you don't have create a new instance of a SQLiteDatabase every time you save an Image